resolver/src/hosts.rs

```diff
@@ -0,0 +1,166 @@
+//! Hosts result from a configuration of `/etc/hosts`
+
+use std::collections::HashMap;
+use std::io::{self, BufRead, BufReader};
+use std::fs::File;
+use std::net::{Ipv4Addr, Ipv6Addr};
+use std::str::FromStr;
+use std::path::Path;
+
+use trust_dns::rr::{Name, RData};
+
+/// Configuration for the local `/etc/hosts`
+#[derive(Debug, Default, Clone)]
+pub struct Hosts {
+    /// Name -> RDatas map
+    pub by_name: HashMap<Name, Vec<RData>>,
+}
+
+impl Hosts {
+    /// Creates a new configuration from /etc/hosts, only works for unix like OSes,
+    /// others will return empty configuration
+    pub fn new() -> Hosts {
+        read_hosts_conf("/etc/hosts").unwrap_or_default()
+    }
+
+    /// lookup_static_host looks up the addresses for the given host from /etc/hosts.
+    pub fn lookup_static_host(&self, name: &Name) -> Option<Vec<RData>> {
+        if self.by_name.len() > 0 {
+            if let Some(val) = self.by_name.get(name) {
+                return Some(val.to_vec());
+            }
+        }
+        None
+    }
+}
+
+#[cfg(unix)]
+pub fn read_hosts_conf<P: AsRef<Path>>(path: P) -> io::Result<Hosts> {
+    let mut hosts = Hosts {
+        by_name: HashMap::new(),
+    };
+
+    let file = File::open(path)?;
+
+    for line in BufReader::new(file).lines() {
+        let line = line.unwrap_or_default();
```

Comment:
can you put a comment here for what the expected input looks like, for reference?

//    ip  name
//    ::1 localhost

Just for reference when looking at the code.

Reply:
ok